tests/test_cmake_build/CMakeLists.txt

```diff
@@ -55,6 +55,14 @@ function(pybind11_add_build_test name)
   add_dependencies(test_cmake_build test_build_${name})
 endfunction()
 
+# Avoid uninitialized variables
+if(NOT DEFINED PYTHON_MODULE_EXTENSION)
+  set(PYTHON_MODULE_EXTENSION "")
+endif()
```

Comment:
Worth turning this into a function, and just write single lines like `set_uninitialized(PYTHON_MODULE_EXTENSION "")`?

Reply:
(because 1. CMake if-statments are quite some clutter, and 2. it's slightly clearer in intention and wouldn't need the comment anymore, but I can see there's a tradeoff with maintenance of that function)

Reply:
Ehh, we could add it, as long the places we need it are not standalone.  How about `possibly_uninitialized` or `touch`?

Reply:
Those two could work as well, yes, if you always want to set them to `""`?

Also, just a suggestion. Feel free to ignore it if you're not convinced; I'm not that advanced in CMake to foresee the consequences. I just know enough to know it's possible ;-)

Reply:
Okay, added `possibly_uninitialized(PYTHON_MODULE_EXTENSION Python_INTERPRETER_ID)` here.

You'll always want to define them to "" since if they are uninitialized, that's what they act like. :)

Reply:
Great, thanks for the explanation.